### PR TITLE
New option to Include deal KVPs when enableSendAllBids === false

### PR DESCRIPTION
**Pull request by robertrmartinez**
## Type of change

- [X] Bugfix

## Description of change
Fix for #3899 

New option in `targetingControls` to allow sending through bids which have dealId's even when enableSendAllBids is false

```
pbjs.setConfig({
   enableSendAllBids: false,
   targetingControls: {
      alwaysIncludeDeals: true
   }
});
```

`alwaysIncludeDeals` must be explicitly set = true to utilize.

This code only happens when `enableSendAllBids` is `false` (as if true should have been added anyways!)

### CODE REVIEW

File: test/spec/unit/core/targeting_spec.js [hunk 1]
```diff
@@ -330,6 +330,73 @@ describe('targeting tests', function () {
...
+
+      let targetingWithAdnxsDeal = {
+        hb_deal_rubicon: '1234',
+        hb_deal: '1234',
+        hb_pb: '0.53',
+        hb_adid: '148018fe5e',
+        hb_bidder: 'rubicon',
+        foobar: '300x250',
+        hb_deal_appnexus: '1234',
+        hb_pb_appnexus: '0.53',
```
Comment (bretg) [inline test/spec/unit/core/targeting_spec.js]: why isn't this 0.1 to match bid4.cpm?  Also, I would find it less confusing to use different deal IDs for RP and APN

File: test/spec/unit/core/targeting_spec.js [hunk 2]
```diff
@@ -330,6 +330,73 @@ describe('targeting tests', function () {
...
+        hb_adid_appnexus: '148018fe5e',
+        hb_bidder_appnexus: 'appnexus',
+      };
+      let targetingWithoutAdnxsDeal = {
```
Comment (bretg) [inline test/spec/unit/core/targeting_spec.js]: I'd also like to see a scenario where the winning bid isn't a deal.

File: test/spec/unit/core/targeting_spec.js (hunk 1 above)
Comment (robertrmartinez) [reply test/spec/unit/core/targeting_spec.js]: Good catch. Let me fix

File: test/spec/unit/core/targeting_spec.js (hunk 2 above)
Comment (robertrmartinez) [reply test/spec/unit/core/targeting_spec.js]: Ok Will do!

**Comment by robertrmartinez** [comment]
@bretg Updated the tests

**Comment by benjaminclot** [comment]
That's great! Now all that's missing is support of this by the [Universal Creative](https://github.com/prebid/prebid-universal-creative)...

**Pull request merged**